File: 2026-merge-triplets-to-form-target-triplet/merge-triplets-to-form-target-triplet.java
class Solution {
    public boolean mergeTriplets(int[][] triplets, int[] target) {
        int px = 0, py = 0, pz = 0;
        for (int[] t : triplets) {
            if (t[0] <= target[0] && t[1] <= target[1] && t[2] <= target[2]) {
                px = Math.max(px, t[0]);
                py = Math.max(py, t[1]);
                pz = Math.max(pz, t[2]);
            }
        }
        return px == target[0] && py == target[1] && pz == target[2];
    }
}

File: 2026-merge-triplets-to-form-target-triplet/merge-triplets-to-form-target-triplet.py
'''
    2   5   3
    1   7   5    
'''

class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        mx, my, mz = target
        px, py, pz = 0, 0, 0
        for x, y, z in triplets:
            if x <= mx and y <= my and z <= mz:     # otherwise discard cur triplet
                px, py, pz = max(px, x), max(py, y), max(pz, z)
            if px == mx and py == my and pz == mz:
                return True

        return False